R/classicProcess.R

```diff
@@ -373,64 +382,75 @@ ClassicProcess <- function(jaspResults, dataset = NULL, options) {
-  modW         <- modelOptions[["modelNumberModeratorW"]]
-  modZ         <- modelOptions[["modelNumberModeratorZ"]]
-
-  # Replace encoded X, W, Z with user variables
-  if (independent != "")
-    vars <- gsub(encoding[["X"]], independent, vars)
-  if (modW != "")
-    vars <- gsub(encoding[["W"]], modW, vars)
-  if (modZ != "")
-    vars <- gsub(encoding[["Z"]], modZ, vars)
-
-  # Is var a mediator?
-  isMed <- grepl(encoding[["M"]], vars)
-  # Which mediator index?
-  medIdx <- stringr::str_extract(vars[isMed], "[0-9]")
-  medIdx <- as.integer(medIdx[!is.na(medIdx)])
-  
-  if (length(medIdx) > 0 && length(mediators) > 0) {
-    for (i in 1:length(medIdx)) {
-      if (length(mediators) >= medIdx[i]) {
-        vars <- gsub(vars[isMed][i], mediators[medIdx[i]], vars)
-      }
-    }
+  independent <-        modelOptions[["modelNumberIndependent"]]
+  mediators   <- unlist(modelOptions[["modelNumberMediators"  ]])
+  modW        <-        modelOptions[["modelNumberModeratorW" ]]
+  modZ        <-        modelOptions[["modelNumberModeratorZ" ]]
+
+  # Apply the JASP coding to X, W, Z, and M if the user has specified the variables and 'vars' still contains the dummy version
+  if (independent != "" && encoding$X %in% vars) {
+    # Apply JASP encoding:
+    jaspCodedX <- encodeColNames(independent)
+    # Sub the JASP-coded name into 'vars':
+    vars       <- gsub(encoding$X, jaspCodedX, vars)
+  }
+
+  if (modW != "" && encoding$W %in% vars) {
+    jaspCodedW <- encodeColNames(modW)
+    vars       <- gsub(encoding$W, jaspCodedW, vars)
+  }
+
+  if (modZ != "" && encoding$Z %in% vars) {
+    jaspCodedZ <- encodeColNames(modZ)
+    vars       <- gsub(encoding$Z, jaspCodedZ, vars)
   }
 
-  # If mediator has no index still replace
-  if ((length(medIdx) == 0) && sum(isMed) > 0) {
-    for (i in 1:length(vars[isMed])) {
-      if (length(mediators) >= i) {
-        vars <- gsub(vars[isMed][i], mediators[i], vars)
+  if (length(mediators) > 0 && any(grepl(encoding$M, vars))) {
+    jaspCodedM <- encodeColNames(mediators)
+
+    ## Find the mediators in 'vars':
+    meds <- grep(paste0("^", encoding$M, "\\d?$"), vars, value = TRUE)
+
+    ### Which mediator index?
+    medIdx <- stringr::str_extract(meds, "[0-9]")
+    medIdx <- as.integer(medIdx[!is.na(medIdx)])
+
+    if (length(medIdx) > 0) {
+      for (i in seq_along(medIdx)) {
+        if (length(mediators) >= medIdx[i])
+          vars <- gsub(meds[i], jaspCodedM[medIdx[i]], vars)
       }
+    } else { # If mediator has no index still replace
+      for (i in seq_along(meds))
+        vars <- gsub(meds[i], jaspCodedM[i], vars)
     }
   }
-  
-  # Replace encoded Y with user variable
-  vars <- gsub(encoding[["Y"]], globalDependent, vars)
 
-  return(encodeColNames(vars))
+  # Apply the JASP encoding to Y if 'vars' still contains the dummy version:
+  if(encoding$Y %in% vars)
+    vars[vars %in% encoding$Y] <- encodeColNames(globalDependent)
```

Comment:
```suggestion
    vars[vars %in% encoding$Y] <- globalDependent
```

The global dependent variable doesn't need to be encoded here because it is automatically encoded by the QML widget (it is an `AssignedVariablesList`). The double encoding for some reason leads to the encoded variable name showing up in the path plot, so I would simply remove it since it's not necessary:

![image](https://github.com/jasp-stats/jaspProcess/assets/63114869/b7fb4d00-e81a-4a8e-84cf-71d4fab63d8d)